File: app/src/main/java/com/cbr/behancesampleapp/network/BehanceApiService.java
package com.cbr.behancesampleapp.network;

import com.example.dimitrios.disample.model.BehanceUser;
import com.example.dimitrios.disample.model.BehanceUserResponse;

import java.util.List;
import java.util.Map;

import io.reactivex.Observable;
import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

/**
 * Created by dimitrios on 24/08/2017.
 */

public interface BehanceApiService {

    @GET(Urls.Behance.Users.GET_USERS)
    Observable<Response<List<BehanceUser>>> getUsers(@QueryMap Map<String, Object> queryParams);

    @GET(Urls.Behance.Users.GET_USER_BY_ID)
    Observable<Response<BehanceUser>> getUserById(@QueryMap Map<String, Object> queryParams);

    @GET(Urls.Behance.Users.GET_USER_BY_ID)
    Observable<Response<BehanceUser>> getUserByUsername(@QueryMap Map<String, Object> queryParams);

}

File: domain/src/main/java/com/futureworkshops/domain/network/BehanceApiService.kt
package com.futureworkshops.domain.network


import com.futureworkshops.data.model.remote.BehanceProject
import com.futureworkshops.data.model.remote.BehanceUser
import com.futureworkshops.domain.network.ListResponse
import com.futureworkshops.domain.network.Urls
import com.futureworkshops.domain.network.Urls.PARAM_API_KEY
import com.futureworkshops.domain.network.Urls.PARAM_ID
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import retrofit2.http.QueryMap

/** Created by dimitrios on 24/08/2017. */
interface BehanceApiService {

    @GET(Urls.GET_USERS)
    fun getUsers(@QueryMap queryParams: @JvmSuppressWildcards Map<String, Any>): Single<ListResponse<BehanceUser>>

    @GET(Urls.GET_USER_BY_ID)
    fun getUserById(@Path(PARAM_ID) id: String, @Query(PARAM_API_KEY) apiKey: String): Single<BehanceUser>

    @GET(Urls.GET_PROJECTS)
    fun getProjects(@QueryMap queryParams: Map<String, Any>): Single<ListResponse<BehanceProject>>

}